crates/sui-bridge/src/types.rs

```diff
@@ -1,27 +1,162 @@
+pub enum BridgeEvent {
+    Sui(SuiBridgeEvent),
+    Eth(EthBridgeEvent),
+}
+
+impl BridgeEvent {
+    /// Convert to message bytes that are verified in Move and Solidity
+    pub fn to_bytes(&self) -> Vec<u8> {
+        let mut bytes = Vec::new();
+        // Add prefix
+        bytes.extend_from_slice(BRIDGE_MESSAGE_PREFIX);
+        match self {
+            BridgeEvent::Sui(SuiBridgeEvent::SuiToEthTokenBridgeV1(e)) => {
+                // Add message type
+                bytes.push(BridgeEventType::TokenTransfer as u8);
+                // Add message version
+                bytes.push(1u8);
+                // Add nonce
+                bytes.extend_from_slice(&e.nonce.to_le_bytes());
+                // Add source chain id
+                bytes.push(BridgeChainId::Sui as u8);
+                // Add source tx id length
+                bytes.push(SUI_TX_DIGEST_LENGTH as u8);
+                // Add source tx id
+                bytes.extend_from_slice(e.sui_tx_digest.as_ref());
+                // Add source tx event index
+                bytes.push(e.sui_tx_event_index as u8);
+
+                // Add source address length
+                bytes.push(SUI_ADDRESS_LENGTH as u8);
+                // Add source address
+                bytes.extend_from_slice(&e.sui_address.to_vec());
+                // Add dest chain id
+                bytes.push(BridgeChainId::Eth as u8);
+                // Add dest address length
+                bytes.push(EthAddress::len_bytes() as u8);
+                // Add dest address
+                bytes.extend_from_slice(e.eth_address.as_bytes());
+
+                // Add token id
+                bytes.push(e.token_id as u8);
+
+                // Add token amount
+                bytes.extend_from_slice(&e.amount.to_le_bytes());
+            }
+            BridgeEvent::Eth(_e) =>
+            // TODO add formats for other events
+            {
+                unimplemented!()
+            }
+        }
+        bytes
+    }
+
+    pub fn keccak256_hash(&self) -> [u8; 32] {
+        let mut hash_function = Keccak256::default();
+        let bytes = self.to_bytes();
+        hash_function.update(&bytes);
+        hash_function.finalize().into()
+    }
 }
 
 pub struct BridgeCommitteeValiditySignInfo {
-    pub signatures: HashMap<BridgeCommitteePublicKey, BridgeCommitteeSignature>,
+    pub signatures: HashMap<BridgeAuthorityPublicKey, BridgeAuthoritySignature>,
+}
+
+pub type SignedBridgeEvent = Envelope<BridgeEvent, BridgeAuthoritySignInfo>;
+pub type VerifiedSignedBridgeEvent = VerifiedEnvelope<BridgeEvent, BridgeAuthoritySignInfo>;
+
+#[derive(Debug, Clone, Copy, Serialize, Deserialize)]
+pub struct BridgeEventDigest(Digest);
+
+impl BridgeEventDigest {
+    pub const fn new(digest: [u8; 32]) -> Self {
+        Self(Digest::new(digest))
+    }
+}
+
+impl Message for BridgeEvent {
+    type DigestType = BridgeEventDigest;
+
+    // this is not used today
```

Comment:
Do you mean that we do not include that intent in the messages, right? I was confused when I first read it, maybe define a generic `IntentScope::Unused` or `IntentScope::UnusedBridgeEvent`